lib/types.d.ts

```diff
@@ -0,0 +1,151 @@
+import {IsEqual} from 'type-fest';
+
+/**
+ * Represents a merged object between `TObject` and `TOtherObject`.
+ * If both objects have a property with the same name, the type from `TOtherObject` is used.
+ */
+type MergeBy<TObject, TOtherObject> = Omit<TObject, keyof TOtherObject> & TOtherObject;
+
+/**
+ * Represents a deeply nested record. It maps keys to values,
+ * and those values can either be of type `TValue` or further nested `DeepRecord` instances.
+ */
+type DeepRecord<TKey extends string | number | symbol, TValue> = {[key: string]: TValue | DeepRecord<TKey, TValue>};
+
+/**
+ * Represents type options to configure all Onyx methods.
+ * It's a combination of predefined options with user-provided options (CustomTypeOptions).
+ *
+ * The options are:
+ * - `keys`: Represents a string union of all Onyx normal keys.
+ * - `collectionKeys`: Represents a string union of all Onyx collection keys.
+ * - `values`: Represents a Record where each key is an Onyx key and each value is its corresponding Onyx value type.
+ *
+ * The user-defined options (CustomTypeOptions) are merged into these predefined options.
+ * In case of conflicting properties, the ones from CustomTypeOptions are prioritized.
+ */
```

Comment:
probably mention that user-defined option is provided through module argumentation?